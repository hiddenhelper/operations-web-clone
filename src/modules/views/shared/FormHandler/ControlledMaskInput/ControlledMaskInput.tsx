import React, { memo } from 'react';
import MaskedInput from 'react-text-mask';

export interface IMaskInputProps {
  inputRef: (ref: any) => void;
  mask: any;
  placeholderChar: string;
  showMask: boolean;
}

const ControlledMaskInput = ({ inputRef, ...props }: IMaskInputProps) => {
  return (
    <MaskedInput
      {...props}
      ref={ref => {
        inputRef(ref ? ref.inputElement : null);
      }}
    />
  );
};

export default memo(ControlledMaskInput);
