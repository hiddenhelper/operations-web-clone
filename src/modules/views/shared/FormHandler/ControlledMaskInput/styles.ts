import { createStyles, makeStyles } from '@material-ui/core/styles';
import { STYLE } from '../../../../../constants';

export const useStyles = makeStyles(theme =>
  createStyles({
    formControl: {
      width: '100%',
      '&:not(:first-child)': {
        margin: theme.spacing(1, 0, 0, 0),
      },
    },
    requiredMark: {
      color: STYLE.COLOR.ERROR,
    },
  })
);
