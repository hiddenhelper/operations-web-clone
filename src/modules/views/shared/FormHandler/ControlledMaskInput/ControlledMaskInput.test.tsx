import React from 'react';
import { render } from '@testing-library/react';
import ControlledMaskInput, { IMaskInputProps } from './ControlledMaskInput';

describe('Controlled Mask Input', () => {
  let props: IMaskInputProps = {
    inputRef: jest.fn(),
    mask: [/\d/, /\d/, '-', /\d/, /\d/, /\d/, /\d/, /\d/, /\d/, /\d/],
    placeholderChar: '\u2000',
    showMask: true,
  };

  it('should render with mask', () => {
    const wrapper = render(<ControlledMaskInput {...props} />);
    expect(wrapper.findByText('  -         '));
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should render without mask', () => {
    props.showMask = false;
    const { container } = render(<ControlledMaskInput {...props} />);
    expect(container).toMatchSnapshot();
  });
});
