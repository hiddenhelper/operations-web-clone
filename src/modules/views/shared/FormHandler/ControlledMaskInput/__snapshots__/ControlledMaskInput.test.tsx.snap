// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controlled Mask Input should render with mask 1`] = `
<div>
  <input
    value=""
  />
</div>
`;

exports[`Controlled Mask Input should render without mask 1`] = `
<div>
  <input
    value=""
  />
</div>
`;
